slam-libraries/viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -319,28 +329,47 @@ bool SLAMServiceImpl::ExtractPointCloudToBuffer(std::stringbuf &buffer) {
 void SLAMServiceImpl::ProcessData() {
     // Set up and build the MapBuilder
     SetUpMapBuilder();
+    DetermineActionMode();
+
+    if (action_mode == SLAMServiceActionMode::UPDATING ||
+        action_mode == SLAMServiceActionMode::LOCALIZING) {
+        // Check if there is an apriori map in the path_to_map directory
+        std::vector<std::string> map_filenames =
+            viam::io::ListSortedFilesInDirectory(path_to_map);
+
+        bool found_map = false;
+        std::string latest_map_filename;
+        for (size_t i = map_filenames.size() - 1; i == 0; i--) {
+            if (map_filenames.at(i).find(".pbstream") != std::string::npos) {
+                latest_map_filename = map_filenames.at(i);
+                found_map = true;
+                break;
+            }
+        }
+        if (!found_map) {
+            throw std::runtime_error(
+                "cannot find maps but they should be present");
+        }
+        bool optimize = true;
+        bool load_frozen_trajectory = true;  // true for LOCALIZING action mode
```

Comment:
`bool load_frozen_trajectory = action_mode == SLAMServiceActionMode::LOCALIZING`

Reply:
Note: maybe dont need to even create the variable if the check can be done inside `LoadFromFile`